File: string/most-common-word/most-common-word.go
package mostcommonword

import (
	"regexp"
	"strings"
)

func mostCommonWord(paragraph string, banned []string) string {
	re := regexp.MustCompile(`[!?',;.]`)
	paragraph = re.ReplaceAllString(paragraph, " ")
	paragraph = strings.ToLower(paragraph)

	ban := make(map[string]bool)
	for _, b := range banned {
		ban[b] = true
	}

	ans := ""
	max := -1
	lookup := make(map[string]int)
	for _, word := range strings.Fields(paragraph) {
		if ban[word] {
			continue
		}
		lookup[word]++
		if lookup[word] > max {
			max = lookup[word]
			ans = word
		}

	}

	return ans

}

File: string/most-common-word/most-common-word.ts
function mostCommonWord(paragraph: string, banned: string[]): string {
    let ans = '';
    let max = -1;

    const lookup = new Set<string>(banned)
    const count: { [key: string]: number } = {};

    paragraph = paragraph.replace(/[!?',;.]/g, ' ').toLowerCase();

    for (let word of paragraph.split(/\s/)) {
        if (lookup.has(word)) {
            continue;
        }
        count[word] = count[word] ? count[word] + 1 : 1;
        if (count[word] > max) {
            max = count[word];
            ans = word;
        }
    }

    return ans;
};
